### ISSUE TITLE
SelectBoxEditor with ModelDataSource cannot be set to "null"

### ISSUE THREAD

**Issue by bweinzierl** [neos/neos-ui#2553]
### Description

When using the SelectBoxEditor with a implementation of AbstractModelDataSource like it is the default for type `Neos\Media\Domain\Model\Tag` the selected option cannot be unselected again after the changes where applied (see screencast).

### Steps to Reproduce

1. Configure Nodetype with a tag property:
```
filterTag:
      type: 'Neos\Media\Domain\Model\Tag'
      defaultValue: null
      ui:
        label: 'Filter by Tag'
        reloadIfChanged: TRUE
        inspector:
          group: 'filterOptions'
          position: 30
          editorOptions:
            dataSourceAdditionalData:
              firstEmpty: true
```
2. Select a Tag in the backend and apply changes 
3. Try to deselect the tag again

#### Expected behavior

It should be possible to deselect the option / select a null value

#### Actual behavior

It switches back to the currently saved option. So the property cannot be set to `null` again.

[Screencast from Slack Channel](https://neos-project.slack.com/files/U050FRJ0S/FMV9ZCSSW/screen_recording_2019-09-11_at_09.22.35.mov)

### Affected Versions

Neos: 
4.3.4 

UI:
3.8.1

**Comment by grebaldi** [comment]
This is still reproducible on 7.3 <s>plus an additional oddity:

![Peek 2023-05-02 20-11](https://user-images.githubusercontent.com/2522299/235750218-dae6dd5a-3819-4f69-b42b-8c6112ae8353.gif)

*(Notice the missing hover effect - those options are supposed to turn blue on hover)*</s>
**EDIT:** This was a hiccup on my system - I was apparently using a stale build of the UI, no clue how that happened ¯\_(ツ)_/¯

Another thing:
> [...]  using the SelectBoxEditor with a implementation of AbstractModelDataSource like it is the default for type `Neos\Media\Domain\Model\Tag` [...]

Is this supposed to be the default? Because: There's no such data source or configuration :confused:

**Comment by grebaldi** [comment]
Should be fixed with: https://github.com/neos/neos-ui/pull/3482
